src/storage/src/hummock/local_version.rs

```diff
@@ -120,18 +216,27 @@ impl LocalVersion {
                         .rev() // Important: order by epoch descendingly
                         .map(|(_, shared_buffer)| shared_buffer.clone())
                         .collect();
+                    let result_sync: Vec<SyncUncommittedData> = guard
+                        .sync_uncommitted_data
+                        .iter()
+                        .filter(|&node| {
+                            node.0 <= read_epoch && node.0 >= smallest_uncommitted_epoch
+                        })
+                        .map(|(_, value)| value.clone())
```

Comment:
Can we possibly avoid `clone` here and use reference in `ReadVersion`?

Reply:
Same for `shared_buffer`

Reply:
We must use clone because we have only one lock in localversion, So we need to use clone to avoid owning the lock until the read is complete